src/ape_ethereum/provider.py

```diff
@@ -1375,14 +1383,35 @@ def stream_request(self, method: str, params: Iterable, iter_path: str = "result
         payload = {"jsonrpc": "2.0", "id": 1, "method": method, "params": params}
         results = ijson.sendable_list()
         coroutine = ijson.items_coro(results, iter_path)
-        resp = requests.post(uri, json=payload, stream=True)
+        resp = requests_lib.post(uri, json=payload, stream=True)
         resp.raise_for_status()
 
         for chunk in resp.iter_content(chunk_size=2**17):
             coroutine.send(chunk)
             yield from results
             del results[:]
 
+    def batch_requests(self, requests: list[dict]) -> Any:
+        if not (uri := self.http_uri):
+            raise ProviderError("This provider has no HTTP URI and is unable to batch requests.")
+
+        for idx, request in enumerate(requests):
+            if "jsonrpc" not in request:
+                request["jsonrpc"] = "2.0"
+            if "id" not in request:
+                request["id"] = idx + 1
```

Comment:
- enumerate has a second arg for start index
- if we allow id overrides (any good reason for that?), then we need to check there are no duplicates